crates/wasmtime/src/module.rs

```diff
@@ -1042,6 +1042,34 @@ impl Module {
         self.inner.memory_images()?;
         Ok(())
     }
+
+    /// Get the map from `.text` section offsets to Wasm binary offsets for this
+    /// module.
+    ///
+    /// Each entry is a (`.text` section offset, Wasm binary offset) pair.
+    ///
+    /// Entries are yielded in order of `.text` section offset.
+    pub fn address_map<'a>(&'a self) -> impl Iterator<Item = (u32, Option<u32>)> + 'a {
```

Comment:
I think it might be good to explain a bit in the docs what the `Option<u32>` is here, namely it's an instruction that doesn't map to any known source location in the original wasm file. I think we use this to avoid corresponding instructions to places they didn't come from but I sort of forget as well.

Also IIRC I think that the map here is based here on ranges where if you get `(1, Some(2))` then `(5, Some(4))` then that means bytes 1, 2, 3, and 4 come from source position 2 and bytes 5 to the next range come from source position 4. Basically we're not going to have an entry per instruction but rather for ranges of instructions (which is what the `None` is used for, to terminate a range).

Finally the first element of the pair returned here might be good to change to a `usize` to assist with indexing into `.text()`?

Oh and one more, for the docs, I think it would be good to mention that the address map is not guaranteed to be available as there's a `Config` option which can disable it.

Reply:
Yeah I can add this stuff to the docs.